src/test/java/org/apache/maven/plugins/jar/AbstractJarMojoTest.java

```diff
@@ -0,0 +1,51 @@
+package org.apache.maven.plugins.jar;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test for {@link AbstractJarMojo}
+ */
+public class AbstractJarMojoTest
+{
+    @Test
+    public void testInvalidModuleNames()
+    {
+        assertFalse( AbstractJarMojo.isValidModuleName( "" ) );
+        assertFalse( AbstractJarMojo.isValidModuleName( "." ) );
+        assertFalse( AbstractJarMojo.isValidModuleName( "_" ) );
+        assertFalse( AbstractJarMojo.isValidModuleName( "dash-is-invalid" ) );
+        assertFalse( AbstractJarMojo.isValidModuleName( "new.class" ) );
```

Comment:
What about some test with ':' and '+'?

Can a module name start with a digit ?
And non latin-1 chars?

Reply:
Will update those and more samples later today.